monitorcontrol/monitorcontrol.py

```diff
@@ -135,6 +159,160 @@ def _get_vcp_feature(self, code: vcp.VCPCode) -> int:
+                if val == "":
+                    continue
+
+                val = int(val, 16)
+                if sub_key is None:
+                    result_dict[val] = []
+                else:
+                    result_dict[sub_key].append(val)
+
+                start = i + 1  # +1 to skip the space
+            if caps_str[i] in "(":
+                sub_key = int(caps_str[start:i], 16)
+                result_dict[sub_key] = []
+                start = i + 1  # +1 to skip the (
+            if caps_str[i] in ")":
+                val = caps_str[start:i]
+
+                if val == "":
+                    continue
+
+                val = int(val, 16)
+                result_dict[sub_key].append(val)
+
+                sub_key = None
+                start = i + 1  # +1 to skip the )
+        return result_dict
+
+    def _parse_capabilities(self, caps_str: str) -> dict:
+        """
+        Converts the capabilities string into a nice dict
+        """
+        caps_dict = {
+            # Used to specify the protocol class
+            "prot": "",
+            # Identifies the type of display
+            "type": "",
+            # The display model number
+            "model": "",
+            # A list of supported VCP codes. Somehow not the same as "vcp"
+            "cmds": "",
+            # A list of supported VCP codes with a list of supported values
+            # for each nc code
+            "vcp": "",
+            # undocumented
+            "mswhql": "",
+            # undocumented
+            "asset_eep": "",
+            # MCCS version implemented
+            "mccs_ver": "",
+            # Specifies the window, window type (PIP or Zone) safe area size
+            # (bounded safe area) maximum size of the window, minimum size of
+            # the window, and window supports VCP codes for control/adjustment.
+            "window": "",
+            # Alternate name to be used for control
+            "vcpname": "",
+            # Parsed input sources into text. Not part of capabilities string.
+            "inputs": "",
+        }
+
+        for key in caps_dict:
+            if key in ["cmds", "vcp"]:
+                caps_dict[key] = self._convert_to_dict(
+                    self._extract_a_cap(caps_str, key)
+                )
+            else:
+                caps_dict[key] = self._extract_a_cap(caps_str, key)
+
+        # Parse the input sources into a text list for readability
+        input_source_cap = vcp.VCPCode("input_select").value
+        if input_source_cap in caps_dict["vcp"]:
+            caps_dict["inputs"] = []
+            input_val_list = caps_dict["vcp"][input_source_cap]
+            input_val_list.sort()
+
+            for val in input_val_list:
+                caps_dict["inputs"].append(InputSource(val).name)
+
+        return caps_dict
+
+    def get_vcp_capabilities(self) -> dict:
+        """
+        Gets the capabilities of the monitor
```

Comment:
```suggestion
        Gets the capabilities of the monitor.
```

What does the output dictionary look like?  Would be good to add an example.

```
Returns:
    VCP capabilities.

    Example return value::

        {"foo": "bar"}
```

Reply:
Hows this:

`        """
        Gets the capabilities of the monitor

        Returns:
            Dictionary of capabilities in the following format:
            {'prot': 'monitor',
             'type': 'LCD',
             'cmds': {
                        1: [],
                        2: [],
                        96: [15, 17, 18],
                     }
             'inputs': ['DP1', 'HDMI1', 'HDMI2']
            }
        """`